File: 프로그래머스/lv2/68645. 삼각 달팽이/삼각 달팽이.swift
import Foundation

func solution(_ n: Int) -> [Int] {
    var answer = [Int]()
    var tri = Array(repeating: Array(repeating: 0, count: n), count: n)
    
    var (x, y) = (-1, 0)
    var num = 1
    
    for i in 0..<n {
        for _ in i..<n {            
            switch i % 3 {
                case 0:
                    x += 1
                case 1:
                    y += 1
                default: // case 2
                    x -= 1
                    y -= 1
            }
            tri[x][y] = num
            num += 1
        }
    }
    
    for i in 0..<n {
        for j in 0...i {
            answer.append(tri[i][j])
        }
    }
    
    return answer
}

File: 프로그래머스/lv2/68645. 삼각 달팽이/삼각 달팽이.py
def solution(n):
    answer = []
    tri = [[0 for _ in range(n)] for _ in range(n)]

    x, y = -1, 0
    num = 1
    for i in range(n):
        for j in range(i, n):
            
            if i % 3 == 0:
                x += 1
            
            elif i % 3 == 1:
                y += 1
            
            elif i % 3 == 2:
                x -= 1
                y -= 1
                
            tri[x][y] = num
            num += 1

    for i in range(n):
        for j in range(i+1):
            answer.append(tri[i][j])
    
    return answer